unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Maroon.Physics;
+using Maroon.Physics.HuygensPrinciple;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InterferencePlate : MonoBehaviour
+{
+    [SerializeField]
+    private SlitPlate slitPlate;
+
+    private float waveLength;
+    private float distanceBetweenSlits;
+    private float distanceBetweenPlates;
+
+    private MeshRenderer meshRenderer;
+
+    void Start()
+    {
+        slitPlate = GameObject.Find("Plate").GetComponent<SlitPlate>();
+        meshRenderer = GetComponent<MeshRenderer>();
+        UpdateParameters();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        //Check if there is a other way to check if distance betwen plates changed
+        if (slitPlate.transform.hasChanged)
```

Comment:
It would be more elegant if we create a UnityEvent OnMovement in the SlitPlate Script for handling this.

Reply:
This will be changed in another issue.